OMEdit/OMEdit/OMEditGUI/Annotations/TextAnnotation.cpp

```diff
@@ -300,96 +300,88 @@ void TextAnnotation::drawTextAnnotaion(QPainter *painter)
   applyLinePattern(painter);
   /* Don't apply the fill patterns on Text shapes. */
   /*applyFillPattern(painter);*/
-  qreal dx = ((-boundingRect().left()) - boundingRect().right());
-  qreal dy = ((-boundingRect().top()) - boundingRect().bottom());
-  // first we invert the painter since we have our coordinate system inverted.
-  painter->scale(1.0, -1.0);
-  painter->translate(0, dy);
+  // store the existing transformations
+  const QTransform painterTransform = painter->transform();
+  const qreal scaleX = painterTransform.m11();
+  const qreal scaleY = painterTransform.m22();
+  const qreal shearX = painterTransform.m21();
+  const qreal shearY = painterTransform.m12();
+  // set new transformation for the text based on rotation or scale.
+  qreal sx, sy;
+  if (painterTransform.type() == QTransform::TxRotate) {
+    // if no flip or XY flip
+    if ((shearX >= 0 && shearY >= 0) || (shearX < 0 && shearY < 0)) {
+      painter->setTransform(QTransform(painterTransform.m11(), 1.0 * (shearY < 0 ? -1.0 : 1.0), painterTransform.m13(),
```

Comment:
One more point. I do not understand why we multiply by 1.0. Should not be any difference here no? or I might just be tired

Reply:
Yeah that's right. I just made it complicated for no reason :). I fixed it now.